smallfile_wrapper/trigger_smallfile.py

```diff
@@ -0,0 +1,114 @@
+from copy import deepcopy
+import time
+import os
+import sys
+import json
+import subprocess
+import logging
+
+
+class SmallfileWrapperException(Exception):
+    pass
+
+class _trigger_smallfile:
+    """
+        Will execute with the provided arguments and return normalized results for indexing
+    """
+    def __init__(self, logger, operations, yaml_input_file, cluster_name, working_dir, result_dir, user, uuid, sample):
+        self.logger = logger
+        self.operations = operations
+        self.yaml_input_file = yaml_input_file
+        self.working_dir = working_dir
+        self.result_dir = result_dir
+        self.user = user
+        self.uuid = uuid
+        self.sample = sample
+        self.cluster_name = cluster_name
+
+    def ensure_dir_exists(self, directory):
+        if not os.path.exists(directory):
+            os.mkdir(directory)
+
+    def emit_actions(self):
+        """
+        Executes test and calls document parsers, if index_data is true will yield normalized data
+        """
+
+        self.ensure_dir_exists(self.working_dir)
+        rsptime_dir = os.path.join(self.working_dir, 'network_shared')
+
+        # clear out any unconsumed response time files in this directory
+        if os.path.exists(rsptime_dir):
+            contents = os.listdir(rsptime_dir)
+            for c in contents:
+                if c.endswith('.csv'):
+                    os.unlink(os.path.join(rsptime_dir, c))
+
+        #execute for each job in the user specified job file
+        operation_list = self.operations.split(',')
+        for operation in operation_list:
+
+            json_output_file = os.path.join(self.result_dir, '%s.json' % operation)
+            network_shared_dir = os.path.join(self.working_dir, 'network_shared')
+            rsptime_file = os.path.join(network_shared_dir, 'stats-rsptimes.csv')
+            cmd = ["python", "smallfile_cli.py", 
+                    "--operation", operation, 
+                    "--top", self.working_dir, 
+                    "--output-json", json_output_file,
+                    "--response-times", "Y",
+                    "--yaml-input-file", self.yaml_input_file ]
+            self.logger.info('running:' + ' '.join(cmd))
+            self.logger.info('from current directory %s' % os.getcwd())
+            try:
+                process = subprocess.check_call(cmd, stderr=subprocess.STDOUT)
+            except subprocess.CalledProcessError as e:
+                self.logger.exception(e)
+                raise SmallfileWrapperException(
+                    'smallfile_cli.py non-zero process return code %d' % e.returncode)
+            self.logger.info("completed sample {} for operation {} , results in {}".format(
+                        self.sample, operation, json_output_file))
+            with open(json_output_file) as f:
+                data = json.load(f)
+                yield data, '-results'
```

Comment:
Do you've an example json document @bengland2 so that we can see what's the structure of the document.

Reply:
[here you go](http://perf1.perf.lab.eng.bos.redhat.com/pub/bengland/tmp/example-smallfile.json)